registration_ref/app.py

```diff
@@ -2,27 +2,37 @@
 from time import sleep
 from typing import Optional
 
-from flask import Flask, abort, jsonify, request
+from flask import Flask, abort, jsonify, request, Response
 import requests
+import logging
 
 from registration_ref.crypto import sign_device_csr
 from registration_ref.sota_toml import sota_toml_fmt
 from registration_ref.settings import Settings
 
 app = Flask(__name__)
 
+logging.basicConfig(level=logging.DEBUG)
+gunicorn_logger = logging.getLogger('gunicorn.access')
+app.logger.handlers = gunicorn_logger.handlers
+app.logger.setLevel(gunicorn_logger.level)
```

Comment:
Why is it important to write to the gunicorn access log?
Isn't it enough to just do basic config and create a regular logger?

Reply:
What do you mean a regular log? What do you suggest instead?
Also, what would be the cons/issues in use gunicorn instead?

Reply:
If you run in gunicorn, what we do, this is easier. Later versions do not really nicely work with Flask/Werkzeug builtin logging. Something to do with gunicorn not capturing STDOUT/STDERR by default correctly. Another option to enable the builtin to work is to append the following flags to running gunicorn: `--capture-output --enable-stdio-inheritance`. That should also make logging possible from any place.

Reply:
At first glance this looks like overengineering.
But, at second glance I realized that your answers create even more questions.

So, the biggest question is this: ***where does this log go?***

Normally, in container world all logs should go to stdout, so that I could do things like `kubectl logs ...` or `docker-compose logs ...`.
For that, I'd expect all apps to just print to stdout and we're done.
That way we don't need to bother if gunicorn captures the log or not - all goes to the same place.

This is achieved as simply as `logging.basicConfig(level=logging.DEBUG); log = logging.getLogger(__module__)`.
That's what I mean by regular logger; just like we do elsewhere.

As I understand, a problem here is the desire to have both gunicorn and regular logs in one place.
That's OK, and is easily achieveable by the `--access-logfile -` flag to gunicorn.
Mind me, we don't even necessarily run under gunicorn - `docker_run.sh` may run plain flask as well.

Reply:
I am totally okay with "normal" logging like you suggested. I just added the reasoning of why one would use that specific logger.

Reply:
THANK YOU a lot for all explanations !!!
I try to keep it simple as suggested since it allow us get the errors and figure out what is wrong already